src/main/java/com/mraof/minestuck/item/StructureScannerItem.java

```diff
@@ -21,65 +37,112 @@ public class StructureScannerItem extends Item
 {
 	private final TagKey<Structure> structure;
 	private final Supplier<Item> fuelItem;
+	private final String scannerName;
 	
-	public StructureScannerItem(Properties properties, TagKey<Structure> structure, Supplier<Item> fuelItem)
+	public StructureScannerItem(Properties properties, TagKey<Structure> structure, Supplier<Item> fuelItem, String scannerName)
 	{
 		super(properties);
 		this.structure = structure;
 		this.fuelItem = fuelItem;
+		this.scannerName = scannerName;
+		
+		/*
+		ItemProperties.register(this, new ResourceLocation("angle"),
+				new CompassItemPropertyFunction((level, stack, entity) -> stack.hasTag() &&
+						stack.getTag().contains("TargetLocation") ? GlobalPos.of(level.dimension(),
+						NbtUtils.readBlockPos(stack.getTag().getCompound("TargetPos"))) : null));
+		*/
+		
+		ItemProperties.register(this, new ResourceLocation("angle"),
+				new CompassItemPropertyFunction((level, stack, entity) -> {
+					if(stack.hasTag() && stack.getTag().contains("TargetLocation"))
+					{
+						return GlobalPos.of(level.dimension(), NbtUtils.readBlockPos(stack.getTag().getCompound("TargetLocation")));
+					} else
+					{
+						return null;
+					}
+				}));
+		
+		ItemProperties.register(this, new ResourceLocation(Minestuck.MOD_ID, "powered"),
+				((pStack, pLevel, pEntity, pSeed) -> pStack.hasTag() && pStack.getTag().getBoolean("Powered") ? 1 : 0));
 	}
 	
 	@Override
-	public InteractionResultHolder<ItemStack> use(Level levelIn, Player playerIn, InteractionHand handIn)
+	public InteractionResultHolder<ItemStack> use(Level pLevel, Player pPlayer, InteractionHand pUsedHand)
```

Comment:
why rename the parameters?

Reply:
Because it overrides the use() function in the Item class. I changed the parameter names to the names used in the original function.

Reply:
Consistency with Parchment parameter names is not as important as just having sensible names. Personally, I always avoid copying the "p_" or "_In" from from such parameter names, however it's not particularly important and shouldn't block the PR from being merged IMO.

Reply:
Alright then. I'll leave it as is, for the time being.